plugins/actionlog/joomla/src/Extension/Joomla.php

```diff
@@ -619,7 +619,37 @@ public function onUserAfterSave($user, $isnew, $success, $msg): void
             'accountlink' => 'index.php?option=com_users&task=user.edit&id=' . $userId,
         ];
 
+        // Check if block / unblock comes from Actions on list
+        if ($task === 'block' || $task === 'unblock') {
+            $messageLanguageKey = 'PLG_ACTIONLOG_JOOMLA_USER_UNBLOCK';
+            $action             = 'unblock';
+            if ($task === 'block') {
+                $messageLanguageKey = 'PLG_ACTIONLOG_JOOMLA_USER_BLOCK';
+                $action             = 'block';
+            }
+
+            $message['action'] = $action;
+        }
+
         $this->addLog([$message], $messageLanguageKey, $context, $userId);
+
+        // Check if on save a block / unblock has changed
+        if ($action === 'update') {
+            $session = Factory::getSession();
```

Comment:
```suggestion
            $session = $this->getApplication()->getSession();
```
This should fix the failing system tests.